### PR TITLE
[console] Fix bug to support console.assert()

### PR DESCRIPTION
**Pull request by cuiyanx**
fix issue #620 

Signed-off-by: Cui, Yan yanx.cui@intel.com

### CODE REVIEW

File: src/zjs_console.c [hunk 1]
```diff
@@ -253,33 +253,36 @@ static jerry_value_t console_assert(const jerry_value_t function_obj,
...
 {
     char message[MAX_STR_LENGTH];
-    uint8_t has_message = 0;
     if (!jerry_value_is_boolean(argv[0])) {
         ERR_PRINT("invalid parameters\n");
         return ZJS_UNDEFINED;
     }
-    bool b = jerry_get_boolean_value(argv[0]);
-    if (!b) {
+    if (argv[1]) {
```
Comment (grgustaf) [inline src/zjs_console.c]: You can't do this, you don't even know if there *is* a second argument yet. It's an optional argument, and even if it wasn't you still have to make sure it's there.

File: src/zjs_console.c [hunk 2]
```diff
@@ -253,33 +253,36 @@ static jerry_value_t console_assert(const jerry_value_t function_obj,
...
         if (argc > 1) {
             if (!jerry_value_is_string(argv[1])) {
                 ERR_PRINT("message parameter must be a string\n");
                 return ZJS_UNDEFINED;
             } else {
                 jerry_size_t sz = MAX_STR_LENGTH;
-                zjs_copy_jstring(argv[0], message, &sz);
+                zjs_copy_jstring(argv[1], message, &sz);
```
Comment (grgustaf) [inline src/zjs_console.c]: OK you're right here, this was a bug, reading the wrong argument.

Comment (grgustaf) [inline src/zjs_console.c]: It's better to use zjs_alloc_from_jstring here so you don't have to worry about the string size as much.

File: src/zjs_console.c [hunk 3]
```diff
@@ -253,33 +253,36 @@ static jerry_value_t console_assert(const jerry_value_t function_obj,
...
+    }
+    for (int i = 0; i < argc; i++) {
+        if (i) {
+            // insert spaces between arguments
+            fprintf(stdout, " ");
+        }
+        if (i == 1) {
+            fprintf(stdout, "- ");
         }
+        print_value(argv[i], stdout, true, false);
```
Comment (grgustaf) [inline src/zjs_console.c]: These arguments aren't supposed to be printed, they're supposed be used with format specifiers in the message string like %d. But as I mentioned elsewhere, I don't think we should attempt to support that at this time.

Comment (cuiyanx) [inline src/zjs_console.c]: I think it looks like the assert function on my test case, but it is not. So, Be closed. 
Please look at #630 .

**Pull request closed without merging**